### PR TITLE
Gun equip changes

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

- Gun equip times changed from 2-3 seconds down to 0.5-1.1 seconds.
- Gun equip slowdown removed.
- Guns will now slow down while held at all times. This slowdown ranges from 0.4 to 0.6

## Why It's Good For The Game

While the gun equip changes did what they were meant to do, people really don't like them and I probably should have changed them sooner but have been busy. This change makes guns provide a slowdown which will still make them worse in melee combat, but not quite as debilitating.
Unfortunately, this means that there is no risk reward to carry a gun and be prepared while risk losing it, however it still makes it easier for people to run away from someone chasing them down with a gun. I'm hoping that while this is a much lighter nerf, it will be enough of a debuff to make guns not quite as powerful in melee combat. If not, the options are either: buff melee weapons, nerf guns, accept it as it is.

To compensate for how slow nukies will be while wielding an LMG, I have reduced the space mode slowdown of the syndicate hardsuit from 1 to 0.5.

The equip time will mainly exist to prevent super quick swapping with hotkeys, as there is a very slight delay which barely affects you if equipping a gun from the holster via the mouse. The slowdowns exist to slightly nerf guns and make it a bit easier for people to hit you in melee combat.

## Testing Photographs and Procedure

https://github.com/BeeStation/BeeStation-Hornet/assets/26465327/e9861482-ace1-478b-8efd-aa22e7615af3

## Changelog
:cl:
balance: Gun equip delay reduced from 2-3 seconds to 0.4-0.8 seconds.
balance: Guns will now provide slight slowdown while equipped.
balance: Syndicate hardsuit slowdown reduced from 1 to 0.5
/:cl:

### CODE REVIEW

**Comment by DarnTheMarn** [comment]
I think everyone can agree the draw timers were a little on the long side, but overall they were very good at their job which was to introduce risk vs. reward for drawing a gun, reduce the effectiveness of guns as a melee weapon, and removing hotkey juggling to prevent yourself from being disarmed. I don't believe this change will help. 

0.4-0.8 is such a small delay that guns will again become the best melee weapons in game but will still prevent a bit of holster juggling.

I don't foresee the move speed modifier having much of an effect in actual play. With essentially zero holster delay it'll be easy for a gun wielder to cripple a leg and reduce anyone to their speed since the most common guns disabler/shotgun/357 deal so much damage in a short amount of time. 

The way I would personally do it is remove the holster slowdown entirely and slightly cut back on the holster delay to 1.25-2s while adding a visual or audio cue to tell you when you can fire. That would mean you still have the chance to get the drop on a person with a gun, but they can still catch you if you don't finish the job before they draw.

**Review by Rukofamicom — APPROVED** [review]

**Comment by PowerfulBacon** [comment]
Slowdown is barely noticable in game, might give it a bit of an increase

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -111,12 +113,16 @@
...
 	if (isnull(equip_time))
-		// Light guns: 1.5 second equip time
-		// Medium guns: 2 second equip time
-		// Heavy guns: 2.5 second equip time
-		equip_time = weapon_weight * 5 + 10
-	if(isnull(spread_unwielded))
+		// Light guns: 0.4 second equip time
+		// Medium guns: 0.6 second equip time
+		// Heavy guns: 0.8 second equip time
+		equip_time = weapon_weight * 2 + 2
```
Comment (PowerfulBacon) [inline code/modules/projectiles/gun.dm]:
```suggestion
		// Light guns: 0.5 second equip time
		// Medium guns: 0.8 second equip time
		// Heavy guns: 1.1 second equip time
		equip_time = weapon_weight * 3 + 2
```

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -111,12 +113,16 @@
...
+	if (isnull(spread_unwielded))
 		spread_unwielded = weapon_weight * 10 + 10
+	if (has_weapon_slowdown)
+		if (!slowdown)
+			slowdown = 0.2 + weapon_weight * 0.1
```
Comment (PowerfulBacon) [inline code/modules/projectiles/gun.dm]:
```suggestion
			slowdown = 0.3 + weapon_weight * 0.1
```

**Comment by PowerfulBacon** [comment]
image

Not really sure if the poll has been properly understood tbh, since it makes no sense for people to widely be voting that the equip time should be lower on a PR that makes the equip time almost instant.

**Comment by Rukofamicom** [comment]
To elaborate on the dead imposter, will this affect shaft miners negatively?

**Comment by PowerfulBacon** [comment]
The slowdown is relatively light and I haven't heard any complaints from shaft-miners or seen issues with them. It will reduce their speed slightly while on lavaland which may make fighting megafauna harder.

**Comment by PowerfulBacon** [comment]
Eh fuck it, feedback wasn't negative, it was more a case of not noticing it and I have heard a lot of dislike towards the gun equip system being so long in the past.

**Pull request merged**